proxy/common/pki.py

```diff
@@ -81,87 +56,72 @@ def gen_private_key(
     return run_openssl_command(command, timeout)
 
 
-def gen_public_key(
+def gen_web_public_key(
         public_key_path: str,
         private_key_path: str,
-        private_key_password: str,
         subject: str,
-        alt_subj_names: Optional[List[str]] = None,
```

Comment:
I don't think we can do this. Also, I think have a different notion of how TLS interception works (I will post about it in separate review comment)

Having alt_subj_names in certificates is a requirement per Mac OS new security policies.  See this https://support.apple.com/en-us/HT210176

Reply:
Thx for the link. 
The strange part is that, FF doesnt care of those "rules".

Reply:
It looks like it's a mandatory field because of chrome, and not only because of OSX => https://support.google.com/chrome/a/answer/7391219?hl=en

Reply:
Exactly. SAN is just needed during certificate signing, so my suggestion is to only change that part of the code. So there are 2 steps happening during interception:

1) Generate a CSR for domain
2) Sign CSR with CA

To generate a CSR we also need domain private key. This is what current code does, it generate an interim key (never saved on disk), pipes the output into CSR signing command.

We'll have to add a similar method where CSR can be generated without having to persist private key `(e.g. gen_priv_key, then remove_passphrase, then gen_csr)`.  You can also remove the need of `remove_passphrase` by not encrypting the private key in first place (i.e. don't use a password while generating one, this will require changes to `gen_priv_key` flags as it by default only generated encrypted private key).  Note that, CSR generation is a one time step and valid till 365 days (current default sign validity).

Also, utilities in `pki.py` are not in anyway tied with its user.  Idea is to simply have all PKI utilities at a central place for re-use.  So, I think terms like "web_cert" shouldn't land in `pki.py`, just continue using standard terminology like private / public keys and signed certificates (for method names).

Hence we must also preserve `DEFAULT_CONFIG` for use when needed. We can make ways to take input custom config `(e.g. SAN during signing)`, overriding default config.

I hope this background helps you refactor better :)